src/bicep-registry/README.md

```diff
@@ -0,0 +1,229 @@
+
+The Bicep Registry contribution guide will have you run `brm generate` before submitting your PR. Do not commit the changes from this command back to the FinOps toolkit repo. All other manual changes to the module itself should be made within the FinOps toolkit repo.
+
+If you find a new requirement arises for Bicep Registry onboarding, please update this document and associated scripts. Please [file an issue](https://github.com/microsoft/cloud-hubs/issues) if you identify any bugs or have ideas to improve the process.
+
+<br>
+
+## Templating language
+
+Bicep offers a clean solution to define reusable modules, but has a few limitations that make it onerous for modules that support multiple scopes. We developed a very simple templating language to work around the following limitations:
+
+1. Modules cannot target multiple scopes.
+2. Scope functions behave differently for each targeted scope.
+3. No mechanism for conditional logic based on scope.
+
+Our templating language supports "scope directives" that enable you to:
+
+1. Leverage existing Bicep developer tooling for the primary scope.
+2. Target multiple scopes.
+3. Conditionally include a single line.
+4. Conditionally include a multiple lines.
+
+### Scope directives
+
+Scope directives are single-line comments that reference one or more Bicep-supported scopes. Scopes are prefixed with `@` and are separated by spaces. Do not include additional characters after the scope directive.
+
+The following example references only the resource group scope:
+
+```bicep
+// @resourceGroup
+```
+
+The following example references all scopes (note order is not important):
+
+```bicep
+// @resourceGroup @subscription @managementGroup @tenant
+```
+
+Refer to the sections below for how to use scope directives.
+
+### Targeting multiple scopes
+
+Modules are designed to work with a single scope. To target multiple scopes, include a scope directive at the top of the file followed by an empty line. Both lines are removed during the build process.
+
+The following example tells the build script to generate modules targeted at resource groups, subscriptions, and management groups:
+
+```bicep
+// @resourceGroup @subscription @managementGroup
+
+@description('The name of the object.')
+param name string
+```
+
+### Conditional lines
+
+Conditional lines are used to include (or exclude) a single line within a Bicep module. To use a conditional line, add a scope directive to the end of the desired line.
+
+The following example includes 2 conditional lines to represent allowed values that are only supported for specific scopes. `value1` is allowed for resource groups and subscriptions, and `value3` is allowed for tenant deployments:
+
+```bicep
+@allowed([
+  'value1' // @resourceGroup @subscription
+  'value2'
+  //'value3' // @tenant
```

Comment:
yeah, the // before the scope when the line is commented is confusing, is it possible to use a different character sequence? //'value3' || @tenant

Reply:
We could, but then that means we have 2 conventions for the same pattern, right? Isn't that more confusing? Fwiw, we don't need the `//`. We could simplify it a couple of ways:
```bicep
// @tenant: 'value3'
// 'value3' @tenant
```
Only reason I like it as-is would be consistency with line 180. Both 180 and 182 are the same, minus one being commented out to ensure valid Bicep code.